packages/admin/cms-admin/src/userPermissions/UserGrid.tsx

```diff
@@ -52,13 +60,45 @@ export const UserPermissionsUserGrid = ({ toolbarAction, rowAction, actionsColum
             headerName: "",
             sortable: false,
             filterable: false,
-            type: "actions",
+
             align: "right",
             pinned: "right",
-            width: actionsColumnWidth,
             disableExport: true,
             renderCell: (params) => {
-                return <> {rowAction && rowAction(params)}</>;
+                const isCurrentUser = params.row.id === currentUser.id;
+                return (
+                    <>
+                        <Tooltip
+                            title={
+                                isCurrentUser ? (
+                                    <FormattedMessage id="comet.userPermissions.cannotImpersonate" defaultMessage="You can't impersonate yourself" />
+                                ) : (
+                                    <FormattedMessage id="comet.userPermissions.impersonate" defaultMessage="Impersonate" />
+                                )
+                            }
+                        >
+                            {/* span is needed for the tooltip to trigger even if the button is disabled*/}
+                            <span>
+                                <IconButton
+                                    disabled={isCurrentUser && !isImpersonated}
+                                    onClick={() => {
+                                        !isCurrentUser && startImpersonation(params.row.id.toString());
+                                        isCurrentUser && isImpersonated && stopImpersonation();
+                                    }}
+                                >
+                                    <ImpersonateUser />
+                                </IconButton>
+                            </span>
+                        </Tooltip>
+                        <IconButton
+                            onClick={() => {
+                                stackApi.activatePage("edit", params.id.toString());
+                            }}
+                        >
+                            <Edit color="primary" />
```

Comment:
When used in an `IconButton`, please always set the color on the `IconButton` instead of the icon itself. 
Otherwise the button's `:hover`, `:active`, `:focus` states will not be in the correct color. 

```suggestion
                            }}
                            color="primary"
                        >
                            <Edit />
```

Reply:
aeea3a2ac115dca495e22ee2bb9ace726128f964